featurebyte/query_graph/node/generic.py

```diff
@@ -468,6 +469,56 @@ def _derive_sdk_code(
         return [], ExpressionStr(expression)
 
 
+class ForwardAggregateNode(BaseNode):
+    """
+    ForwardAggregateNode class.
+    """
+
+    class Parameters(BaseModel):
+        """
+        Forward aggregate parameters
+        """
+
+        horizon: Optional[str]
+        blind_spot: Optional[str]
+        keys: List[InColumnStr]  # group by keys
+        parent: Optional[InColumnStr]
+        agg_func: AggFunc
+        value_by: InColumnStr  # aggregate column
+        serving_name: str
+        entity_ids: Optional[List[PydanticObjectId]]
+        # TableDetails are needed because we will be performing the aggregation query directly on the calling table.
+        table_details: TableDetails
```

Comment:
Should we inherit `featurebyte.query_graph.node.mixin.BaseGroupbyParameters`? If not, we need to update the logic in entity extractor to consider target  (https://github.com/featurebyte/featurebyte/blob/main/featurebyte/query_graph/transform/entity_extractor.py#L34-L49). Entity extractor is used to extract entity(s) of the feature (i suspect we likely need to reuse the same logic for target).

Reply:
also inherited the `AggregationOpStructMixin` to avoid having to duplicate the derive operation structure info if that seems reasonable!